docs/designs/link-output.md

```diff
@@ -0,0 +1,64 @@
+# Link And Bookmarks Output
+
+Docfx generate `links` information of each files into one single file `.links.json` for future data utilization of downstream service of docfx.
+
+## Links
+
+`links` includes all the links which has been processed by docfx:
+
+- Relative link in source content will be processed by docfx and convert it to absolute link if it's valid, otherwise keep it untouched.
+
+- Absolute link in source content will NOT processed by docfx
+
+| Links in source content | source content examples | Output in `.links.json` |
+| ----------------------- | ----------------------- | ---------------------- |
+| Relative Link           | ./a.md (valid)          | /basepath/article/a    |
+|                         | a/b.md (invalid)        | a/b.md                 |
+| Absolute Link           | /basepath/articles/b    | /basepath/articles/b   |
+|                         | www.github.com/x        | www.github.com/x       |
+|                         | https://docs.com/y      | https://docs/com/y     |
+
+> NOTE: the absolute links converted from relative links by docfx don't contains `locale` and `versioning` information
+
+## Output
+
+`.links.json` file will be generated by docfx to contains all the links information about all the page files in build.
+
+*output data model*:
+
+  |            |                       |
+  |----------- |-----------------------|
+  | `links`    | Array of *link model* |
+
+*link model*:
+
+  |                         |                      |
+  |-------------------------|----------------------|
+  | `source_url`            | /dotnet/articles/a   |
+  | `source_moniker_group`  | 218c13d0             |
+  | `target_url`            | /dotnet/articles/b   |
+
+Example:
+
+```json
+{
+    "links":
+    [
+        {
+            "source_url": "/dotnet/articles/a",
+            "source_moniker_group": "218c13d0",
+            "target_url": "/dotnet/articles/b"
+        },
+        {
+            "source_url": "/dotnet/articles/a",
+            "source_moniker_group": "218c13d0",
```

Comment:
Can you give me the moniker name list too, we want to ping the page with moniker.